src/Chapter4.hs

```diff
@@ -629,7 +667,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM  mbA mbB = mbA >>= (\a -> if a == False then pure a else mbB >>= (\b -> pure $ b && a))
```

Comment:
That's the correct solution 👍🏻 
You can simplify it a little bit like this:

```suggestion
andM  ma mb = ma >>= \a -> if a then mb else pure False
```

Reply:
Woah,  I don't even need to use another (>>=) and  (&&) operator in the new solution.